internal/dnfjson/cache.go

```diff
@@ -114,81 +114,94 @@ func newRPMCache(path string, maxSize uint64) *rpmCache {
+
+		// See updateInfo NOTE on error handling
+		cacheEntries, _ := os.ReadDir(path)
+
+		var totalSize uint64
+
+		// Collect the paths grouped by their repo ID
+		// We assume the first 64 characters of a file or directory name are the
+		// repository ID because we use a sha256 sum of the repository config to
+		// create the ID (64 hex chars)
+		for _, entry := range cacheEntries {
+			eInfo, err := entry.Info()
+			if err != nil {
+				// skip it
+				continue
+			}
 
-	// Collect the paths grouped by their repo ID
-	// We assume the first 64 characters of a file or directory name are the
-	// repository ID because we use a sha256 sum of the repository config to
-	// create the ID (64 hex chars)
-	for _, entry := range cacheEntries {
-		eInfo, err := entry.Info()
-		if err != nil {
-			// skip it
-			continue
-		}
+			fname := entry.Name()
+			if len(fname) < 64 {
+				// unknown file in cache; ignore
+				continue
+			}
+			repoID := fname[:64]
+			repo, ok := repos[repoID]
+			if !ok {
+				// new repo ID
+				repoIDs = append(repoIDs, repoID)
+			}
+			mtime := eInfo.ModTime()
+			ePath := filepath.Join(path, entry.Name())
+
+			// calculate and add entry size
+			size, err := dirSize(ePath)
+			if err != nil {
+				// skip it
+				continue
+			}
+			repo.size += size
+			totalSize += size
 
-		fname := entry.Name()
-		if len(fname) < 64 {
-			// unknown file in cache; ignore
-			continue
-		}
-		repoID := fname[:64]
-		repo, ok := repos[repoID]
-		if !ok {
-			// new repo ID
-			repoIDs = append(repoIDs, repoID)
-		}
-		mtime := eInfo.ModTime()
-		ePath := filepath.Join(path, entry.Name())
+			// add path
+			repo.paths = append(repo.paths, ePath)
 
-		// calculate and add entry size
-		size, err := dirSize(ePath)
-		if err != nil {
-			// skip it
-			continue
+			// if for some reason the mtimes of the various entries of a single
+			// repository are out of sync, use the most recent one
+			if repo.mtime.Before(mtime) {
+				repo.mtime = mtime
+			}
+
+			// update the collection
+			repos[repoID] = repo
 		}
-		repo.size += size
-		totalSize += size
 
-		// add path
-		repo.paths = append(repo.paths, ePath)
+		//
+		// update global cache info
+		//
 
-		// if for some reason the mtimes of the various entries of a single
-		// repository are out of sync, use the most recent one
-		if repo.mtime.Before(mtime) {
-			repo.mtime = mtime
+		r.size += totalSize
+		// TODO: replace loop with maps.Copy() introduced in Go 1.21
+		for k, v := range repos {
+			r.repoElements[k] = v
 		}
-
-		// update the collection
-		repos[repoID] = repo
 	}
+
 	sortFunc := func(idx, jdx int) bool {
 		ir := repos[repoIDs[idx]]
 		jr := repos[repoIDs[jdx]]
 		return ir.mtime.Before(jr.mtime)
 	}
 
-	// sort IDs by mtime (oldest first)
 	sort.Slice(repoIDs, sortFunc)
 
-	r.size = totalSize
-	r.repoElements = repos
-	r.repoRecency = repoIDs
+	// sort IDs by mtime (oldest first)
+	r.repoRecency = append(r.repoRecency, repoIDs...)
```

Comment:
`repoRecency` is nilled at the start of this function and is only referenced again here at the end.  Is there a reason we can't do `r.repoRecency = repoIDs`?